### PR TITLE
[FIX] Allow no location in app

### PR DESCRIPTION
**Pull request by benjaminkostiuk**
## Overview

Allow the app to work even if location is turned off by users. This is to not block the apple store approve process.

Updates the app to either use default values, not show the location distance calculations, or removes the functionality altogether. Allows users to press out of the popup asking them to enable location.

This change is a

- [x] Bug fix
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that will change existing functionality)
- [ ] Documentation update

## Description

 - Add a new button to opt out of turning on location
 - Add checks and defaults to all api calls requiring location
 - Refactor default location to null instead of 0 to allow for checks if the location has been updated.
 - Refactor code to either not use location (like in map) or only show distance calculations if we have their location.

## Motivation/Links
Allow the apple app store review process to go through.

### CODE REVIEW

File: src/hooks/checkLocationPermission.tsx [hunk 1]
```diff
@@ -39,8 +39,16 @@ export const requestPermissions = () => {
       {
         text: "Open Settings",
         style: "default",
+        isPreferred: true, // @ts-ignore This is only available on iOS
```
Comment (dhruv423) [inline src/hooks/checkLocationPermission.tsx:42]: This doesnt seem to make a visual change on my end

**Comment by dhruv423** [comment]
We need to change OnboardingScreen.tsx line 119 to bypass if location is not enabled, we should first prompt the screen but the second time we should let them thru, rn it just bricks them there. Also we need to change line 19, again prompt them once and then after let them slide

File: src/hooks/checkLocationPermission.tsx (hunk 1 above)
Comment (benjaminkostiuk) [reply src/hooks/checkLocationPermission.tsx:42]: It was in the docs but I guess it doesn't work 🤣. Will remove

**Comment by benjaminkostiuk** [comment]
> We need to change OnboardingScreen.tsx line 119 to bypass if location is not enabled, we should first prompt the screen but the second time we should let them thru, rn it just bricks them there. Also we need to change line 19, again prompt them once and then after let them slide

Whoops, forgot about onboarding. Will fix.

**Review by dhruv423 — APPROVED** [review]
Thank you for fixing this quickly!

**Pull request merged**